### PR TITLE
[OP#48106] use same component as project tab for searching in smart picker

### PR DESCRIPTION
**Pull request by SwikritiT**
Previously we used the default search provider for the smart picker but we this PR we define the custom smart picker so that the search is similar to the one we have implemented in the project tab.

## Demonstration

https://github.com/nextcloud/integration_openproject/assets/41103328/b8e2a927-1728-4b45-a84d-fcee513501f0

### CODE REVIEW

File: src/components/tab/SearchInput.vue [hunk 1]
```diff
@@ -170,13 +194,20 @@ export default {
...
-						workPackage.fileId = fileId
-						workPackage = await workpackageHelper.getAdditionalMetaData(workPackage)
-						const alreadyLinked = this.linkedWorkPackages.some(el => el.id === workPackage.id)
-						const alreadyInSearchResults = this.searchResults.some(el => el.id === workPackage.id)
-						// check the state again, it might have changed in between
-						if (!alreadyInSearchResults && !alreadyLinked && this.isStateLoading) {
-							this.searchResults.push(workPackage)
+						if (this.isSmartPicker) {
+						   workPackage = await workpackageHelper.getAdditionalMetaData(workPackage)
+						   if (this.isStateLoading) {
```
Comment (SagarGi) [inline src/components/tab/SearchInput.vue:199]: is this if again necessary for state loading?

File: src/views/WorkPackagePickerElement.vue [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+		linkedWorkPackages: [],
+		state: STATE.OK,
+		isAdminConfigOk: loadState('integration_openproject', 'admin-config-status'),
+	}),
+
+	computed: {
+	},
+
+	mounted() {
+	},
```
Comment (SagarGi) [inline src/views/WorkPackagePickerElement.vue:75]:
```suggestion
	
```
might be added when needed.

File: src/components/tab/SearchInput.vue (hunk 1 above)
Comment (SwikritiT) [reply src/components/tab/SearchInput.vue:199]: no not needed thanks for pointing it out

**Review by SagarGi — APPROVED** [review]
LGTM :+1:

**Pull request merged**